hippynn/layers/physics.py

```diff
@@ -106,37 +106,36 @@ def __init__(self, energy_conversion_factor, screening, radius=None):
         self.bond_summer = pairs.MolPairSummer()
 
     def forward(self, charges, pair_dist, pair_first, pair_second, mol_index, n_molecules):
-
-        # Calculation of pair terms
-        base_coulomb = charges[pair_first] * charges[pair_second] / pair_dist.unsqueeze(1)
         screening = self.screening(pair_dist, self.radius).unsqueeze(1)
         screening = torch.where((pair_dist < self.radius).unsqueeze(1), screening, torch.zeros_like(screening))
-        coulomb_pairs = base_coulomb * screening
-
-        # Add pair contributions to system
-        coulomb_molecule = self.bond_summer(coulomb_pairs, mol_index, n_molecules, pair_first)
 
-        # Finally, account for symmetry pairs and energy conversion factor.
-        coulomb_molecule = 0.5 * self.energy_conversion_factor * coulomb_molecule
+        # Voltage pairs for per-atom energy
+        voltage_pairs = self.energy_conversion_factor * (charges[pair_second] / pair_dist.unsqueeze(1)) 
+        voltage_pairs = voltage_pairs * screening 
+        n_atoms, _ = charges.shape
+        voltage_atom = torch.zeros((n_atoms, 1), device=charges.device, dtype=charges.dtype)
+        voltage_atom.index_add_(0, pair_first, voltage_pairs) 
+        coulomb_atom = voltage_atom * charges
+        coulomb_molecules = 0.5 * self.summer(coulomb_atom, mol_index, n_molecules)
 
-        return coulomb_molecule
+        return coulomb_molecules, coulomb_atom
```

Comment:
Isn't this double-counting on the atom level? In HEnergy, atom energies are run through the system summation (`self.summer`) to produce the total energy. In this case there is a factor of 0.5 between them.

Reply:
In the new commits, this issue is resolved. 
Coulomb energy is defined for a pair of atoms. Here, we adopt the  convention that the Coulomb energy for a pair of atoms is evenly partitioned to both atoms as the 'per-atom energies'. Therefore, the atom energies sum to the molecular energy; similar to the `HEnergy`.